pkg/resources/postgres_credential.go

```diff
@@ -0,0 +1,218 @@
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "default",
+				Description: "Default schema name",
+			},
+			"username": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Username for Postgres/Redshift/AlloyDB",
+			},
+			"password": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Sensitive:   true,
+				Description: "Password for Postgres/Redshift/AlloyDB",
+			},
+			"num_threads": &schema.Schema{
+				Type:        schema.TypeInt,
+				Optional:    true,
+				Description: "Number of threads to use",
+			},
+		},
+
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+	}
+}
+
+func resourcePostgresCredentialCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*dbt_cloud.Client)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	isActive := d.Get("is_active").(bool)
+	projectId := d.Get("project_id").(int)
+	type_ := d.Get("type").(string)
+	defaultSchema := d.Get("default_schema").(string)
+	targetName := d.Get("target_name").(string)
+	username := d.Get("username").(string)
+	password := d.Get("password").(string)
+	numThreads := d.Get("num_threads").(int)
+
+	postgresCredential, err := c.CreatePostgresCredential(projectId, isActive, type_, defaultSchema, targetName, username, password, numThreads)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(fmt.Sprintf("%d%s%d", postgresCredential.Project_Id, dbt_cloud.ID_DELIMITER, *postgresCredential.ID))
+
+	resourcePostgresCredentialRead(ctx, d, m)
+
+	return diags
+}
+
+func resourcePostgresCredentialRead(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	c := m.(*dbt_cloud.Client)
+
+	// Warning or errors can be collected in a slice type
+	var diags diag.Diagnostics
+
+	projectId, err := strconv.Atoi(strings.Split(d.Id(), dbt_cloud.ID_DELIMITER)[0])
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	postgresCredentialId, err := strconv.Atoi(strings.Split(d.Id(), dbt_cloud.ID_DELIMITER)[1])
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	postgresCredential, err := c.GetPostgresCredential(projectId, postgresCredentialId)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if err := d.Set("credential_id", postgresCredentialId); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("is_active", postgresCredential.State == dbt_cloud.STATE_ACTIVE); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("project_id", postgresCredential.Project_Id); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("default_schema", postgresCredential.Default_Schema); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("username", postgresCredential.Username); err != nil {
+		return diag.FromErr(err)
+	}
+	if err := d.Set("password", postgresCredential.Password); err != nil {
```

Comment:
Yeah so assuming `c. GetPostgresCredential` does not return the password from the API, this will always be set to "" which is what's causing the resource test to fail, if you want to set it during create (and we should add a warning about storing passwords in state files as it seems fairly standard to do so, as a log or in the docs), you'll probably need to access it from the resourceData with `d.get("password").(string)` or something instead?